src/ManagedMod.cs

```diff
@@ -0,0 +1,75 @@
+using Bep = BepInEx;
+using IO = System.IO;
+
+namespace Silksong.DataManager;
+
+internal record ManagedMod(string Guid, IOnceSaveDataMod? OnceSaveData)
+{
+    internal static bool TryCreate(Bep.BaseUnityPlugin plugin, out ManagedMod instance)
+    {
+        instance = null!;
+
+        var onceSaveData = plugin as IOnceSaveDataMod;
+
+        if (onceSaveData is null)
+            return false;
```

Comment:
I guess the idea here is that this would later become the following (and so on for other data types)?

```cs
var onceSaveData = plugin as IOnceSaveDataMod;
var globalSettings = plugin as IGlobalSettingsMod;

if (oneSaveData is null && globalSettings is null)
    return false
```